modules/shared/src/main/scala/org/tessellation/security/Hash.scala

```diff
@@ -33,6 +35,8 @@ object hash {
     def empty: Hash = Hash(s"%064d".format(0))
 
     implicit val arbitrary: Arbitrary[Hash] = Arbitrary(Gen.stringOfN(64, Gen.hexChar).map(Hash(_)))
+
+    implicit val showHash: Show[Hash] = Show.show[Hash](h => s"Hash(${h.value.take(5)}...${h.value.takeRight(5)})")
```

Comment:
this way you changed all existing debug messages so we won't see any full hash

Reply:
Yeah, that's why I was talking about non implicit show instance.

Reply:
@TheMMaciek @marcinwadon
Can you be more specific on what you want? a non-implicit show instance means it has to be specifically invoked. So it's either an object method that takes hash as a parameter or an instance method like I had before, but returning a `Show` instead of `String`.